Tests/person_tests.py

```diff
@@ -56,3 +56,124 @@ def test_cannot_add_person_with_missing_arguments(self):
+
+    def test_can_successfully_reallocate_fellow_from_one_living_space_to_another(self):
+        """ Test successful reallocation of person from one living space to another """
+
+        self.dojo = Dojo()
+        self.dojo.create_room("living", ["Kampala"])
+        self.dojo.create_room("office", ["Kenya"])
+        self.dojo.add_person('Akira', 'Menai', 'FELLOW', 'Y')
+        self.dojo.add_person('rex', 'acey', 'FELLOW', 'Y')
+        self.dojo.create_room("living", ["Nairobi"])
+        self.dojo.reallocate_person("Akira", "Menai", "Nairobi")
+        self.assertTrue(len(self.dojo.total_rooms[0].occupants) == 1)
+        self.assertTrue(len(self.dojo.total_rooms[2].occupants) == 1)
+
+    def test_can_successfully_reallocate_person_on_waiting_list_a_room(self):
+        """ Test successful reallocation of person on waiting list a room  """
+
+        self.dojo = Dojo()
+        self.dojo.add_person('James', 'macavoy', 'FELLOW', 'Y')
+        self.dojo.create_room("office", ["kenya"])
+        self.dojo.create_room("living", ["Kampala"])
+        self.dojo.reallocate_person("James", "macavoy", "kenya")
+        self.dojo.reallocate_person("James", "macavoy", "Kampala")
+        self.assertTrue(len(self.dojo.total_rooms[0].occupants) == 1)
+        self.assertTrue(len(self.dojo.total_rooms[1].occupants) == 1)
+
+    def test_cannot_reallocate_person_if_person_does_not_exist(self):
+        """ Test failure when reallocating person who does not exist """
+
+        self.dojo = Dojo()
+        self.dojo.create_room("office", ["kenya"])
+        self.dojo.add_person('Ian', 'pyr', 'FELLOW', 'Y')
+        msg1 = self.dojo.reallocate_person("James", "macavoy", "kenya")
+        self.assertEquals(msg1, "Sorry that person does not exist")
+
+    def test_cannot_reallocate_person_if_room_does_not_exist(self):
+        """ Test failure when reallocating person to room that does not exist """
+
+        self.dojo = Dojo()
+        self.dojo.create_room("office", ["kenya"])
+        self.dojo.add_person('chris', 'Jaba', 'FELLOW', 'Y')
+        self.dojo.create_room("office", ["Somalia"])
+        msg1 = self.dojo.reallocate_person("chris", "Jaba", "Ghana")
+        self.assertEquals(msg1, "Sorry that person does not exist")
+
+    def test_cannot_reallocate_person_in_room_with_maximum_occupants(self):
+        """ Test failure when reallocating person to room at maximum capacity """
+
+        self.dojo = Dojo()
+        self.dojo.create_room("office", ["kenya"])
+        self.dojo.add_person('Ian', 'pyr', 'FELLOW')
+        self.dojo.add_person('Joe', 'jameson', 'FELLOW')
+        self.dojo.add_person('Ruben', 'konia', 'FELLOW')
+        self.dojo.add_person('James', 'macavoy', 'FELLOW')
+        self.dojo.add_person('Akira', 'Menai', 'FELLOW')
+        self.dojo.add_person('rex', 'acey', 'FELLOW')
+        self.assertTrue(len(self.dojo.total_rooms[0].occupants) == 6)
+        self.dojo.add_person('Larry', 'smith', 'FELLOW')
```

Comment:
In the functionality, i check whether a person is waiting for allocation. if so  he is allocated a room